uwsift/workspace/importer.py

```diff
@@ -715,6 +715,12 @@ def _set_kind_metadata(self, attrs: dict) -> None:
             LOG.info(f"No data kind configured for reader '{self.reader}'." f" Falling back to 'IMAGE'.")
             attrs[Info.KIND] = Kind.IMAGE
 
+        if attrs[Info.KIND] is Kind["IMAGEorPOINTS"]:
+            if isinstance(attrs["area"], SwathDefinition) and len(attrs["area"].shape) == 1:
+                attrs[Info.KIND] = Kind.POINTS
+            else:
+                attrs[Info.KIND] = Kind.IMAGE
```

Comment:
I was thinking/wondering if maybe the "image or points" kind wasn't an actual kind, but more of a special config-only value that is interpreted in this method (or some new one that makes it more clear that it is doing this conversion, outside the importer). So basically this method in the importer always assigns a real `Kind` enum, but the input "kind" from the config could be other special values like "dynamic" or "image_or_points", but those don't actually exist in the `Kind` enum.

Does that make sense?

My worry is that we don't want to add more to `Kind` because we don't want the other parts of SIFT to have to handle that.

Additionally, I think I like the idea of a "dynamic" value rather than a specific "this can only be an image or points" data. The dynamic at-load time behavior would be the same for most if not all cases. It also gets around readers adding new types of data in the future. Thoughts?

Reply:
Good points, thanks for the feedback!

I also thought about the fully-dynamic approach, and I agree it is more elegant (there's the risk we end up with too many permutations), but my initial worry was that we don't have, at the moment, enough metadata in the `attrs` to be able to decide dynamically every time (e.g. between other types combinations). On the other hand, we could implement `DYNAMIC`, and document that atm it handles only the image/point ambiguity with the given assumptions, and more combinations will be supported in the future if needed - this is what I now did.

I refactored the code to not add a kind anymore as well.